testcases/kernel/syscalls/read/read02.c

```diff
@@ -130,5 +138,7 @@ static struct tst_test test = {
 	.test = verify_read,
 	.setup = setup,
 	.cleanup = cleanup,
-	.needs_tmpdir = 1,
+	// needs_tmpdir creates a temporary directory under tmp for testing
```

Comment:
@vamsikrishna935  https://github.com/lsds/sgx-lkl/pull/723 fixed the writing /tmp issue. Please revert tmp related changes as well.

Reply:
But in the latest repo it is failing, in particular to this test, the issue fix is not applicable it seems. With the above changes only the test is passing. otherwise it is failing with above error that you have mentioned in the previous comments.

Reply:
This is a different issue. 

The problem occurs in safe_macros.c line 221.

It returns -1 with an errno of EINVAL.

Either not using /tmp or not using O_DIRECT to open the file and everything passes. As of 2007 at least, this is expected behavior:

https://lists.archive.carbon60.com/linux/kernel/720702

And apparently as of 2019:

https://ask.fedoraproject.org/t/need-direct-io-for-ramfs-or-tmpfs-or-is-there-another-linux-ram-disk-that-supports-direct-io/3528

It appears that not running on /tmp would be the correct thing to do here, however, a much better patch comment is needed to explain the issue.

Reply:
Updated comments.